python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -69,42 +81,69 @@ def list_of_strings(arg):
 
 args, _ = parser.parse_known_args()
 
 if __name__ == "__main__":
     engine_args = None
-    if _vllm and not args.disable_vllm:
+    if args.model_dir:
+        model_id_or_path = Path(Storage.download(args.model_dir))
+    else:
+        model_id_or_path = args.model_id
+
+    if model_id_or_path is None:
+        raise ValueError("You must provide a model_id or model_dir")
+
+    if (
+        vllm_available
+        and not args.disable_vllm
+        and infer_vllm_supported_from_model_architecture(model_id_or_path) is not None
+    ):
+        from .vllm.vllm_model import VLLMModel
+
         args.model = args.model_dir or args.model_id
-        args.revision = args.model_revision
-        engine_args = AsyncEngineArgs.from_cli_args(args)
-    predictor_config = PredictorConfig(
-        args.predictor_host,
-        args.predictor_protocol,
-        args.predictor_use_ssl,
-        args.predictor_request_timeout_seconds,
-    )
-    model = HuggingfaceModel(
-        args.model_name,
-        engine_args=engine_args,
-        predictor_config=predictor_config,
-        kwargs=vars(args),
-    )
+        engine_args = build_vllm_engine_args(args)
+        model = VLLMModel(args.model_name, engine_args)
+
+    else:
+        predictor_config = PredictorConfig(
+            args.predictor_host,
+            args.predictor_protocol,
+            args.predictor_use_ssl,
+            args.predictor_request_timeout_seconds,
+        )
+        task = None
+        kwargs = vars(args)
+        if kwargs.get("task", None) is not None:
+            try:
+                task = MLTask[kwargs["task"]]
+            except KeyError:
+                raise ValueError(f"Unsupported task: {kwargs['task']}")
+
+        model = HuggingfaceModel(
+            args.model_name,
+            model_id_or_path=model_id_or_path,
+            model_revision=kwargs.get("model_revision", None),
+            tokenizer_revision=kwargs.get("tokenizer_revision", None),
+            predictor_config=predictor_config,
+            do_lower_case=not kwargs.get("disable_lower_case", False),
+            add_special_tokens=not kwargs.get("disable_special_tokens", False),
+            max_length=kwargs["max_length"],
+            tensor_input_names=kwargs.get("tensor_input_names", None),
+            return_token_type_ids=kwargs.get("return_token_type_ids", None),
+            task=task,
+        )
     try:
         model.load()
-        kserve.ModelServer().start([model] if model.ready else [])
     except ModelMissingError:
         logging.error(
             f"fail to locate model file for model {args.model_name} under dir {args.model_dir},"
             f"trying loading from model repository."
         )
+    if not args.model_id:
         kserve.ModelServer(
             registered_models=HuggingfaceModelRepository(args.model_dir)
         ).start([model] if model.ready else [])
+    else:
+        kserve.ModelServer().start([model] if model.ready else [])
```

Comment:
are we reverting back the change in https://github.com/kserve/kserve/pull/3424?

Reply:
Fixed. This was unintentional.